apps/site/next.config.js

```diff
@@ -1,4 +1,25 @@
+// All server-side API requests are handled by lib/utils/api.ts
+// but this rewrite still exists for locally testing native POST requests
+const LOCAL_API_URL = "http://localhost:8000";
+
+// When deployed on Vercel, this path acts like a rewrite in vercel.json
+// Next.js would normally be unable to find the API endpoint
+// but Vercel somehow steps in and makes the Serverless Function visible
+const VERCEL_API_PATH = "/api/";
+
 /** @type {import('next').NextConfig} */
-const nextConfig = {}
+const nextConfig = {
+	rewrites: async () => {
+		return [
+			{
+				source: "/api/:path*",
+				destination:
+					process.env.NODE_ENV === "development"
+						? `${LOCAL_API_URL}/:path*`
+						: VERCEL_API_PATH,
```

Comment:
Question: Do we need to append `/:path*` to the end of `VERCEL_API_PATH` as well or is that something we'll save for another issue?

Reply:
I don't fully understand what's going on under the hood, but it appears that Vercel treats the Next rewrite as a [Vercel rewrite](https://vercel.com/docs/edge-network/rewrites), so `/api/` references the Serverless Function defined in `api/index.py`. Including `/:path*` would actually break the rewrite and make other API subpaths inaccessible. The value in `next.config.js` could even be `/api/index.py` and Vercel somehow interprets that correctly.

The rewrite was previously expressed in `vercel.json` but removed since it appeared to be redundant, and a rewrite in `next.config.js` was needed anyway for proxying to localhost and could be included with comments unlike in `vercel.json`.